src/parse.rs

```diff
@@ -677,17 +677,31 @@ fn dlt_payload<T: NomByteOrder>(
     verbose: bool,
     payload_length: u16,
     arg_cnt: u8,
-    is_controll_msg: bool,
+    msg_type: Option<MessageType>,
 ) -> IResult<&[u8], PayloadContent, DltParseError> {
     if verbose {
         match count(dlt_argument::<T>, arg_cnt as usize)(input) {
-            Ok((rest, arguments)) => Ok((rest, PayloadContent::Verbose(arguments))),
+            Ok((rest, arguments)) => {
+                if let Some(MessageType::NetworkTrace(_)) = msg_type {
+                    let slices = arguments
+                        .iter()
+                        .map(|i| match &i.value {
+                            Value::Raw(bytes) => bytes.clone(),
+                            _ => vec![],
+                        })
+                        .filter(|i| !i.is_empty())
```

Comment:
The method `filter_map()` solves this in a slightly more elegant way :nerd_face: 
```rust
                    let slices = arguments
                        .iter()
                        .filter_map(|i| match &i.value {
                            Value::Raw(bytes) => Some(bytes.clone()),
                            _ => None,
                        })
                        .collect();
```

Reply:
Absolutely!